src/skeptic/lib.rs

```diff
@@ -789,3 +794,45 @@ fn test_sanitization_of_testnames() {
     assert_eq!(sanitize_test_name("my_long__fun___name___with____a_____lot______of_______spaces"), "my_long_fun_name_with_a_lot_of_spaces");
     assert_eq!(sanitize_test_name("Löwe 老虎 Léopard"), "l_we_l_opard");
 }
+
+#[test]
+fn line_numbers_displayed_are_for_the_end_of_each_code_block() {
+    let lines = &[
+        "Rust code that should panic when running it.",
+        "",
+        "```rust,should_panic",
+        "fn main() {",
+        "    panic!(\"I should panic\");",
+        "}",//6
+        "```",
+        "",
+        "Rust code that should panic when compiling it.",
+        "",
+        "```rust,no_run,should_panic",
+        "fn add(a: u32, b: u32) -> u32 {",
+        "    a + b",
+        "}",
+        "",
+        "fn main() {",
+        "    add(1);",
+        "}",//18
+        "```",
+    ];
+    let mut old_template = None;
+
+    let tests = extract_tests_from_string(&mut create_test_input(&to_pseudo_file(lines)), &String::from("blah"), &mut old_template);
+
+    let test_names: Vec<String> = tests.into_iter().map(|test| get_line_number_from_test_name(test)).collect();
+
+    assert_eq!(test_names, vec!("6", "18"));
+}
+
+#[cfg(test)]
+fn get_line_number_from_test_name(test: Test) -> String {
+    String::from(test.name.split("_").last().expect("There were no underscores!"))
+}
+
+#[cfg(test)]
+fn to_pseudo_file(lines: &[&str]) -> Vec<String> {
+    lines.iter().map(|line| format!("{}{}", *line, "\n")).collect()
```

Comment:
no need to use two variables `format!("{}{}", *line, "\n")`. why not use just format!("{}\n", *line) 

also when we move from slices of str's into multi-line string literals, I would suggest renaming this function to something more informative like `get_lines`.